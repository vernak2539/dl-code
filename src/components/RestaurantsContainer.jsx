import React from "react";
import Restaurant from "./Restaurant";

export default ({ restaurants }) => {
  if (!restaurants.length) {
    return (
      <div className="section">
        <div className="restaurant-count text-gray hide-when-small">
          Loading...
        </div>
      </div>
    );
  }

  return (
    <div className="section">
      <div className="restaurant-count text-gray hide-when-small">
        {restaurants.length} Restaurants
      </div>
      <ul className="restaurant-container">
        {restaurants.map(restaurant => (
          <li key={restaurant.id}>
            <Restaurant {...restaurant} />
          </li>
        ))}
      </ul>
    </div>
  );
};
