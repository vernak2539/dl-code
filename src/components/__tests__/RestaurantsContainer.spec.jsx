import React from 'react';
import { shallow, render } from 'enzyme';
import RestaurantsContainer from '../RestaurantsContainer';

describe('Component > RestaurantsContainer', () => {
	it('should render correctly', () => {
		const restaurants = [
			{
				id: '63460',
				name: 'Burrito Kitchen',
				image:
					'https://f.roocdn.com/images/menus/25785/header-image.jpg?width=360&height=200&auto=webp&format=jpg&fit=crop&v=1477308353',
				url: 'https://deliveroo.co.uk/menu/london/bank/burrito-kitchen-cheapside',
				price: 2,
				tags: ['Mexican', 'Burritos', 'Halal']
			},
			{
				id: '33406',
				name: 'Pure',
				image:
					'https://f.roocdn.com/images/menus/52101/header-image.jpg?width=360&height=200&auto=webp&format=jpg&fit=crop&v=1515776356',
				url: 'https://deliveroo.co.uk/menu/london/the-city/pure-gracechurch-st',
				price: 2,
				tags: ['Salads', 'Gluten Free', 'Smoothies']
			}
		];

		expect(shallow(<RestaurantsContainer restaurants={restaurants} />)).toMatchSnapshot();
	});

	it('should display loading if data is not present', () => {
		expect(shallow(<RestaurantsContainer restaurants={[]} />)).toMatchSnapshot();
	});
});
