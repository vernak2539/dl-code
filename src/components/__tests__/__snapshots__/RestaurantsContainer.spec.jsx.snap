// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component > RestaurantsContainer should display loading if data is not present 1`] = `
<div
  className="section"
>
  <div
    className="restaurant-count text-gray hide-when-small"
  >
    Loading...
  </div>
</div>
`;

exports[`Component > RestaurantsContainer should render correctly 1`] = `
<div
  className="section"
>
  <div
    className="restaurant-count text-gray hide-when-small"
  >
    2
     Restaurants
  </div>
  <ul
    className="restaurant-container"
  >
    <li
      key="63460"
    >
      <Component
        id="63460"
        image="https://f.roocdn.com/images/menus/25785/header-image.jpg?width=360&height=200&auto=webp&format=jpg&fit=crop&v=1477308353"
        name="Burrito Kitchen"
        price={2}
        tags={
          Array [
            "Mexican",
            "Burritos",
            "Halal",
          ]
        }
        url="https://deliveroo.co.uk/menu/london/bank/burrito-kitchen-cheapside"
      />
    </li>
    <li
      key="33406"
    >
      <Component
        id="33406"
        image="https://f.roocdn.com/images/menus/52101/header-image.jpg?width=360&height=200&auto=webp&format=jpg&fit=crop&v=1515776356"
        name="Pure"
        price={2}
        tags={
          Array [
            "Salads",
            "Gluten Free",
            "Smoothies",
          ]
        }
        url="https://deliveroo.co.uk/menu/london/the-city/pure-gracechurch-st"
      />
    </li>
  </ul>
</div>
`;
